### DISCUSSION TITLE
How does this relate to/interface with RDF?

### DISCUSSION THREAD

**Discussion by hoijui** [terminusdb/terminusdb#830] [Q&A] [answered]
First thing in my mind when I found your product:

* Could we use TerminusDB to store RDF data?
* Is it possible to run SPARQL queries?
* If I use this, am I locked in, or could I change to something else that uses RDF?
* ... or easily move my data from something else to TerminusDB?
* more concrete: could I export TerminusDB content into JSON-LD files stored in a git repo (or import from that)?

**Comment by GavinMendelGleason** [comment, marked as answer]
Hello! I'll try to answer in order:

* Yes - we store all data internally as RDF
* Not without work - WOQL is similar but different to SPARQL - it is possible to implement a SPARQL query engine but we haven't gotten around to it
* You can dump RDF and use elsewhere, or you can dump JSON and use in something like Mongo - so actually you probably have more exit options than most databases.
* See above
* We can ingest JSON-LD and elaborate to JSON-LD internally but we actually don't have a flag to return it to the user at the moment. This is actually a relatively trivial feature addition but isn't there yet.

**Reply by lifan0127** [reply]
Hello Gavin, I see that Terminus DB recently add support for GraphQL. Do you have plan to add SPARQL support as well?

**Reply by GavinMendelGleason** [reply]
We don't have active plans to add SPARQL support, but it would not be a particularly hard project to do so.
